packages/backend-common/api-report.md

```diff
@@ -145,6 +147,29 @@ export interface ContainerRunner {
   runContainer(opts: RunContainerOptions): Promise<void>;
 }
 
+// @public
+export interface Context {
+  readonly abortSignal: AbortSignal_2;
+  readonly deadline: Date | undefined;
+  use(...decorators: ContextDecorator[]): Context;
+  value<T = unknown>(key: string | symbol): T | undefined;
```

Comment:
Perhaps `key: unknown` here as well? although embedding the type within the key is a bit tempting :grin:

It's not too important though, interaction with these values should always be through wrappers imo

Reply:
Leaving the key type a bit restricted is fine I think, wouldn't want to encourage any other things being used really. Agreed that it's a bit tempting with a key type (as in your excellent precursor to this), but I came to the same conclusion - when I experimented with adding some interaction patterns on top of this interface it usually turned out a lot better if one did all of the interaction through wrapper classes with static methods for creation and extraction. So the key type became less interesting then.

Reply:
Yep, true. Probably best to optimize for duplicate package installations instead tbh. We could straight up switch this to just `key: symbol` and recommend usage of `Symbol.for`?

Reply:
Yeah true, only string ain't bad either since it's definitely gonna keep the duplication problem away, and effectively has the same size namespace as the symbols anyway. Only string for now?

Reply:
Decided!